agenda.php

```diff
@@ -9,15 +9,126 @@ class NotImplementedException extends BadMethodCallException {}
 
 require __DIR__ . '/vendor/autoload.php';
 
-interface Agenda {
-    public function __construct(string $url, string $username, string $password, object $api, array $agenda_args);
-    public function getUserEventsFiltered(string $userid, array $filters_to_apply = array());
-    public function parseEvent(string $userid, object $event);
-    public function getEvents();
-    public function update();
-    public function updateAttendee(string $url, string $usermail, bool $add, ?string $attendee_CN=null);
-    public function getEvent(string $url);
-    public function clearEvents();
+abstract class Agenda {
+    protected $caldav_client;
+    protected $api;
+
+    abstract public function __construct(string $vCalendarFilename, string $username, string $password, object $api, array $agenda_args);
+    abstract public function getUserEventsFiltered(string $userid, array $filters_to_apply = array());
+    abstract public function getParsedEvent(string $vCalendarFilename, string $userid);
+    abstract public function getEvents();
+    abstract protected function update(array $ETags);
+    
+    abstract protected function saveEvent(string $vCalendarFilename, string $ETag, object $vCalendar);
+    abstract protected function getEvent(string $vCalendarFilename);
+    abstract public function clearEvents();
+
+    abstract protected function getCTag();
+    abstract protected function setCTag(string $CTag);
+    
+    public function checkAgenda() {
+        $remote_CTag = $this->caldav_client->getCTag();
+        if(is_null($remote_CTag)) {
+            $this->log->error("Fail to update the CTag");
```

Comment:
It's detail, but what about logging "Fail to get ctag from the remote server" (which may be more explicit on what went wrong)?

Reply:
Updated with 0434723